src/NuGet.Core/NuGet.ProjectManagement/Projects/BuildIntegratedNuGetProject.cs

```diff
@@ -182,7 +203,7 @@ public virtual string ProjectName
         {
             get
             {
-                return MSBuildNuGetProjectSystem.ProjectName;
+                return Path.GetFileNameWithoutExtension(MSBuildProjectPath);
```

Comment:
I'd suggest to initialize the `ProjectName` in the constructor. Thus if the actual path is invalid, e.g. empty or containing invalid characters, the error will be detected earlier at the initialization.